File: src/main/java/com/example/store/dto/ResponseMessage.java
package com.example.store.dto;

public class ResponseMessage {
    private Integer status;
    private String message;
    private Object data;

    public ResponseMessage(Integer status, String message, Object data) {
        this.status = status;
        this.message = message;
        this.data = data;
    }

    public ResponseMessage(Integer status, String message) {
        this.status = status;
        this.message = message;
        this.data = null;
    }

    public ResponseMessage(Object data, Integer status) {
        this.status = status;
        this.message = null;
        this.data = data;
    }

    public Integer getStatus() {
        return status;
    }

    public void setStatus(Integer status) {
        this.status = status;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public Object getData() {
        return data;
    }

    public void setData(Object data) {
        this.data = data;
    }
}

File: src/main/kotlin/com/example/store/dto/ResponseMessage.kt
package com.example.store.dto

class ResponseMessage {
    private var status: Int?
    private var message: String?
    private var data: Any?

    constructor(status: Int?, message: String?, data: Any?) {
        this.status = status
        this.message = message
        this.data = data
    }

    constructor(status: Int?, message: String?) {
        this.status = status
        this.message = message
        data = null
    }

    constructor(data: Any?, status: Int?) {
        this.status = status
        message = null
        this.data = data
    }

    fun getStatus(): Int? {
        return status
    }

    fun setStatus(status: Int?) {
        this.status = status
    }

    fun getMessage(): String? {
        return message
    }

    fun setMessage(message: String?) {
        this.message = message
    }

    fun getData(): Any? {
        return data
    }

    fun setData(data: Any?) {
        this.data = data
    }
}